internal/executer/executer.go

```diff
@@ -49,7 +51,14 @@ func New(name string) Executer {
 
 // Execute operates on an executable binary and supports context.
 func (c *executable) Execute(ctx context.Context, input io.Reader, action string) ([]byte, error) {
+	trace := trace.GetTraceHooksFromContext(ctx)
+	if trace != nil {
+		trace.ExecuteStart(c.name, action)
+	}
 	cmd := exec.CommandContext(ctx, c.name, action)
+	if trace != nil {
+		trace.ExecuteDone(c.name, action)
```

Comment:
What if `trace.ExecuteDone` is nil?

Reply:
Added the check. Thanks for catching it.